wallet_ui/components/routes/Dashboard.tsx

```diff
@@ -110,8 +110,13 @@ export function Dashboard(props: { open: boolean; onOpenToggle: () => void }) {
   const classes = useStyles();
 
   const [events, setEvents] = useState<EventList>();
+  const [action, setAction]= useState<object[]>([]);
 
-  React.useEffect(() => {
+  function updateEvent(eventObj : object) {
```

Comment:
I think that this update trigger, plus all the prop drilling it requires, can be avoided. Let's look at this after you've looked at the other comments